src/core/sock/sock-extra.cpp

```diff
@@ -549,7 +549,11 @@ static void xlio_buf_free(struct xlio_buf *buf)
     mem_buf_desc_t *desc = reinterpret_cast<mem_buf_desc_t *>(buf);
     ring_slave *rng = desc->p_desc_owner;
 
-    (void)rng->reclaim_recv_single_buffer(desc);
+    desc->p_next_desc = nullptr;
+    bool ret = rng->reclaim_recv_buffers(desc);
```

Comment:
Not related to this change. What is the idea of keeping the mentioned list of buffers (m_rx_buffs_rdy_for_free_head) ? Should we remove completely?

Reply:
The list reminds me other similar lists in sockets and other places. The idea was likely to accumulate single objects first and do the handling in a batch. However, this is not cache efficient.
I agree that we should handle single object without accumulation, but it need more deeper analysis of the impact.